code/src/App.js

```diff
@@ -1,9 +1,44 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { BrowserRouter, Route, Switch } from 'react-router-dom'
+
+import { MovieList } from './MovieList'
+import { MovieDetails } from './MovieDetails'
+
 
 export const App = () => {
+  const [movies, setMovies] = useState([])
+
+  useEffect(() => {
+    const API_v3_KEY = 'f969c3f19f84733bab447597689e9ea0'
+    const API_URL = `https://api.themoviedb.org/3/trending/movie/week?api_key=${API_v3_KEY}`
+
+    fetch(API_URL)
+      .then((response) => {
+        return response.json()
+      })
+      .then(json => {
+        setMovies(json.results)
+      })
+  }, [])
+
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <BrowserRouter>
+      <Switch>
+        <Route path="/" exact>
+          <MovieList movies={movies} />
+        </Route>
+
+        {
+          movies.map(movie => {
+            return (
+              <Route key={movie.id} path={`/movies/${movie.title}`} exact>
```

Comment:
That's not correct way to do that - instead of generating fixed path for each movie, we should rather use parameter with colon :

```
<Route path="/movies/:movieTitle">
  <MovieDetails />
</Route>
```

Needs to be fixed

Reply:
When I do this it stops working and is just a black page... I don't understand